File: backend/YPU_System/app/Models/Problem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Problem extends Model
{
    use HasFactory;

    protected $fillable = ['doctor_id', 'employee_id', 'description', 'status'];

    public function doctor(): BelongsTo
    {
        return $this->belongsTo(Doctor::class);
    }

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class);
    }
}

File: lib/Model/Problem.dart
class Problem {
  int? id;
  int? doctorId;
  int? employeeId;
  String? description;
  int? status;
  var createdAt;
  var updatedAt;

  Problem(
      {this.id,
      this.doctorId,
      this.employeeId,
      this.description,
      this.status,
      this.createdAt,
      this.updatedAt});

  Problem.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    doctorId = json['doctor_id'];
    employeeId = json['employee_id'];
    description = json['description'];
    status = json['status'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
  }
}
